### DISCUSSION TITLE
Open Source vs. Close Source

### DISCUSSION THREAD

**Discussion by latishasumardy** [albertkun/21S-ASIAAM-191A#119] [Past Thinking Caps]
There are pros and cons of both types of software. For open source software, the biggest pro is that the source code is open to the public. This means that everyone can see it, so the community can help with maintenance and finding/fixing bugs in the code. But a big downside is that because anyone can edit the code, any malicious developers can just as easily introduce bugs/viruses into the software. 

For closed source software, one pro is that it is built with a specific set of users in mind, so the tools are much more fully developed and fleshed out. Another pro is that because you are purchasing the software, the company has the incentive to keep you as a customer, so they provide more customer support and help. One con of closed source software is that because the source code is kept secret, it may be difficult for the user to make changes they need, as they cannot modify the code itself. Another con is that as the company is attempting to keep the code secret, the team working on the code is a lot smaller than communities working on open source software, leading to the possibility of missing bugs or errors in the code.

Reference: [https://brandongaille.com/19-pros-and-cons-of-closed-source-software/]

**Comment by rh5140** [comment]
I like the pros and cons you bring up, particularly how there are more limitations to how closed source software can be changed to streamline personal use. Any changes would need to be within already established systems in the software, which wouldn't have the range to serve all potential needs. I think while companies _are_ incentivized to keep you as a customer and would as a result ensure some baseline level of quality, there might be no motivation to improve their services once they're already profitable. Plus there are a lot of large tech monopolies, so even if some services are lacking, there just might not be other options.

[Wilny's thinking cap](https://github.com/albertkun/21S-ASIAAM-191A/discussions/113#discussion-3328283) also discussed security issues, and the nuances he brought up were super interesting! You should check out what he said and the resources he linked!

**Comment by lvalles10** [comment]
Your comments on company incentive for closed source tools remind me of Salesforce. As a rapidly growing CRM (Salesforce just acquired Slack I believe) it's well-positioned to be the go-to CRM for companies of all scales. I agree with the comment above though that when a company such as Salesforce establishes such a robust customer base and acquires more closed source tools to become the dominant figure of workplace technology, the motivation for service improvements does subside. 

This entire framing of closed source tools as companies makes me think of the ecosystem of Salesforce more broadly. There's Salesforce, its new acquisition of Slack, and entire team of consultants at various firms who work to understand and know Salesforce so they can apply that knowledge in consulting projects.

Most broadly, closed source tools create this problematic ecosystem. As closed source tools become more dominant in the marketplace, they swallow up other close source tools to monopolize certain industries and make it relatively difficult for other companies to launch. This comment then enters a different sphere about the nature of the free market, but for the purposes of innovation and competition it seems as if closed source tools saturate the marketplace and limit beneficial ideation. 

It's interesting to think about a world in which Salesforce was open-source. Such a powerful CRM could be utilized by smaller, local businesses to grow and scale at earlier stages, and the functionality of the CRM could be better tailored to suit their needs. Salesforce already prides itself on fueling community businesses, but to imagine a world where its resources were open source is fascinating to think about. It would be so empowering!
